### PR TITLE
feat: added endpoint to return .json for the input .tiff file

### PR DESCRIPTION
**Pull request by khanjasir90**
Signed-off-by: khanjasir90 <khanmohdjasir@gmail.com>

# Pull Request Template

## Description
This PR creates a /getConfig endpoint which would return the .json file for the corresponding .tiff file.
URL:http://localhost:8080/gcbm/getConfig
Curl : curl -d "title=run4&file_name=Classifier1_moja" -X POST http://localhost:8080/gcbm/getConfig

Fixes # (issue) #183 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)
Output : 
image

...

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have tagged the reviewers in a comment below incase my pull request is ready for a review
- [x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

### CODE REVIEW

**Comment by khanjasir90** [comment]
@Namyalg could u pls take a look at this one!

**Comment by Namyalg** [comment]
Hi @khanjasir90, looks good. 
A few observations : 

1. Running the command with a file that does not exist, throws the error `simulation does not exist`, I think this needs to be changed, the error should report that the file does not exist

image

2. Can you add snapshots, or list some of the other edge cases you have tried?

**Comment by Namyalg** [comment]
CC : @SanjaySinghRajpoot @padmajabhol

File: local/rest_api_gcbm/app.py [hunk 1]
```diff
@@ -350,6 +350,43 @@ def gcbm_dynamic():
...
+            description: Name of the File
+    """
+    # Default title = Simulation
+    title = request.form.get("title") or "simulation"
+    file_name = request.form.get("file_name")
+    input_dir = f"{os.getcwd()}/input/{title}/{file_name}.json"
+
+    # Check if simulation exists or not
+    if not os.path.exists(f"{input_dir}"):
+        return {"error": "Simulation does'nt exist"}, 400
```
Comment (Namyalg) [inline local/rest_api_gcbm/app.py:382]: Change to `Simulation with name " + title + " does not exist"

Comment (khanjasir90) [reply local/rest_api_gcbm/app.py:382]: @Namyalg I'll do the above mentioned change!

**Comment by khanjasir90** [comment]
What if a file which does not have a JSON is supplied, eg : the gcbm_input.db file ?
- Since we are taking the file without any extension then it will throw file does'nt exist for the given simulation.
I'll add some ss related to edge cases asap in.

**Comment by SanjaySinghRajpoot** [comment]
> 

Makes sense, the frontend would read the data from this file (.tiff) with the help of this endpoint. The data will be then displayed to the user and with the another endpoint that is the `/setConfig` the data will be updated on the backend. So if the API is unable to find the file we need to send a specific response message to the frontend.

**Comment by khanjasir90** [comment]
@Namyalg @SanjaySinghRajpoot I have made the specified changes and also added some strict constraints.pls have a look at it and share what are your thoughts on the same. I have also add snapshots of the edge cases for which I have exec the endpoint.
Edge Case 1 :  No Simulation name specified
nosimulation
Edge Case 2:  No File name specified
nofilename
Edge Case 3: Wrong File Name
wrongfilename
EdgeCase 4:  Wrong Simulation Name
wrongsimulation
Correct Output: 
correct

**Review by SanjaySinghRajpoot — DISMISSED** [review]

File: local/rest_api_gcbm/app.py [hunk 2]
```diff
@@ -350,6 +350,56 @@ def gcbm_dynamic():
...
+    if title == "":
+        return {"error": "No Simulation name specified"}, 400
+
+    # check if file_name is empty
+    if file_name == "":
+        return {"error": "No file name specified"}, 400
+
+    # Check if simulation exists or not
+    if not os.path.exists(f"{input_dir}"):
+        return {"error": "Simulation with the name " + title + " does'nt exists"}, 400
```
Comment (Namyalg) [inline local/rest_api_gcbm/app.py:390]: Change to `doesn't`

File: local/rest_api_gcbm/app.py [hunk 3]
```diff
@@ -350,6 +350,56 @@ def gcbm_dynamic():
...
+
+    input_dir_file = f"{input_dir}/{file_name}.json"
+    # Check if file exists or not
+    if not os.path.exists(f"{input_dir_file}"):
+        return {"error": "File with name " + file_name + " does'nt exists"}, 400
```
Comment (Namyalg) [inline local/rest_api_gcbm/app.py:395]: Change to `doesn't`

**Review by Namyalg — COMMENTED** [review]
The various scenarios are handled well.
Thank you @khanjasir90 
Once these minor changes are done, it will be ready for merge!

**Comment by khanjasir90** [comment]
Thanks @Namyalg ! I have made the appropriate changes :)

**Review by Namyalg — APPROVED** [review]

**Pull request merged**